apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -142,6 +153,22 @@ terminate(_Reason, _State) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+reset_state(State) ->
+    case State#state.ref of
+        undefined -> ok;
+        Ref -> ok = httpc:cancel_request(Ref)
+    end,
+    erlang:cancel_timer(State#state.timeout_ref),
+    erlang:cancel_timer(State#state.backoff_timer_ref),
+    State#state{pid = undefined, ref = undefined, size = undefined,
```

Comment:
That kind of reset can cause some issues in the future when a new field will be added to the state, but will not be reset. Can we return just a new `#state{}` here?

Reply:
In that case, can we explicitly set only subscribers in a new `#state{}`? `cancel_timer(FooRef), #state{subs=Subs}.`. All of those default values look not ok here, especially constants like `15000`.